components/event-bus/internal/knative/subscription/controller/eventactivation/reconcile.go

```diff
@@ -3,112 +3,121 @@ package eventactivation
 
 const (
 	finalizerName = "eventactivation.finalizers.kyma-project.io"
 )
 
-type reconciler struct {
-	client   client.Client
-	recorder record.EventRecorder
-	time     util.CurrentTime
-}
+//Reconciler EventActivation reconciler
+type Reconciler struct {
+	// wrapper for core controller components (clients, logger, ...)
+	*reconciler.Base
+
+	// listers index properties about resources
+	eventActivationLister applicationconnectorlistersv1alpha1.EventActivationLister
 
-// Verify the struct implements reconcile.Reconciler
-var _ reconcile.Reconciler = &reconciler{}
+	// clients allow interactions with API objects
+	applicationconnectorClient applicationconnectorclientv1alpha1.ApplicationconnectorV1alpha1Interface
+	kymaEventingClient         eventingclientv1alpha1.EventingV1alpha1Interface
 
-func (r *reconciler) InjectClient(c client.Client) error {
-	r.client = c
-	return nil
+	time util.CurrentTime
 }
 
 // Reconcile reconciles a EventActivation object
-func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	log.Info("Reconcile: ", "request", request)
-
-	ctx := context.TODO()
-	ea := &eventingv1alpha1.EventActivation{}
-	err := r.client.Get(ctx, request.NamespacedName, ea)
-
-	// The EventActivation may have been deleted since it was added to the workqueue. If so, there is
-	// nothing to be done.
-	if errors.IsNotFound(err) {
-		log.Info("Could not find EventActivation: ", "err", err)
-		return reconcile.Result{}, nil
-	}
-
-	// Any other error should be retrieved in another reconciliation. ???
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	ea, err := eventActivationByKey(key, r.eventActivationLister)
 	if err != nil {
-		log.Error(err, "Unable to Get EventActivation object")
-		return reconcile.Result{}, err
+		return err
 	}
 
-	log.Info("Reconciling Event Activation", "UID", string(ea.ObjectMeta.UID))
-
 	// Modify a copy, not the original.
 	ea = ea.DeepCopy()
 
 	// Reconcile this copy of the EventActivation and then write back any status
 	// updates regardless of whether the reconcile error out.
 	requeue, reconcileErr := r.reconcile(ctx, ea)
 	if reconcileErr != nil {
-		log.Error(reconcileErr, "Reconciling EventActivation")
-		r.recorder.Eventf(ea, corev1.EventTypeWarning, "EventactivationReconcileFailed", "Eventactivation reconciliation failed: %v", reconcileErr)
+		r.Recorder.Eventf(ea, corev1.EventTypeWarning, "EventactivationReconcileFailed",
+			"Eventactivation reconciliation failed: %v", reconcileErr)
 	}
 
-	if updateStatusErr := util.UpdateEventActivation(ctx, r.client, ea); updateStatusErr != nil {
-		log.Error(updateStatusErr, "Updating EventActivation status")
-		r.recorder.Eventf(ea, corev1.EventTypeWarning, "EventactivationReconcileFailed", "Updating EventActivation status failed: %v", updateStatusErr)
-		return reconcile.Result{}, updateStatusErr
+	if updateStatusErr := util.UpdateEventActivation(r.applicationconnectorClient, ea); updateStatusErr != nil {
```

Comment:
The name `updateStatusErr` is misleading, this is not a status update. `err` should be good enough here.